modules/programs/termite.nix

```diff
@@ -0,0 +1,365 @@
+        example = "fullscreen = true";
+        type = types.lines;
+        description = "Extra options that should be added to [options] section.";
+      };
+
+      colorsExtra = mkOption {
+        default = "";
+        example = ''
+          color0 = #3f3f3f
+          color1 = #705050
+          color2 = #60b48a
+        '';
+        type = types.lines;
+        description = "Extra colors options that should be added to [colors] section.";
+      };
+
+      hintsExtra = mkOption {
+        default = "";
+        example = "border = #3f3f3f";
+        type = types.lines;
+        description = "Extra hints options that should be added to [hints] section.";
+      };
+    };
+  };
+
+  config = (
+    let
+      boolToString = v: if v then "true" else "false";
+      optionalBoolean = name: val: if val != null then "${name} = ${boolToString val}" else "";
+      optionalInteger = name: val: if val != null then "${name} = ${toString val}" else "";
+      optionalString = name: val: if val != null then "${name} = ${val}" else "";
+    in mkIf cfg.enable {
+      home.packages = [ pkgs.termite ];
+      home.file.".config/termite/config".text = ''
+        [options]
+        ${optionalBoolean "allow_bold" cfg.allowBold}
+        ${optionalBoolean "audible_bell" cfg.audibleBell}
+        ${optionalString "browser" cfg.browser}
+        ${optionalBoolean "clickable_url" cfg.clickableUrl}
+        ${optionalString "cursor_blink" cfg.cursorBlink}
+        ${optionalString "cursor_shape" cfg.cursorShape}
+        ${optionalBoolean "dynamic_title" cfg.dynamicTitle}
+        ${optionalBoolean "filter_unmatched_urls" cfg.filterUnmatchedUrls}
+        ${optionalString "font" cfg.font}
+        ${optionalBoolean "fullscreen" cfg.fullscreen}
+        ${optionalString "geometry" cfg.geometry}
+        ${optionalString "icon_name" cfg.iconName}
+        ${optionalBoolean "modify_other_keys" cfg.modifyOtherKeys}
+        ${optionalBoolean "mouse_autohide" cfg.mouseAutohide}
+        ${optionalBoolean "scroll_on_keystroke" cfg.scrollOnKeystroke}
+        ${optionalBoolean "scroll_on_output" cfg.scrollOnOutput}
+        ${optionalInteger "scrollback_lines" cfg.scrollbackLines}
+        ${optionalString "scrollbar" cfg.scrollbar}
+        ${optionalBoolean "search_wrap" cfg.searchWrap}
+        ${optionalBoolean "size_hints" cfg.sizeHints}
+        ${optionalBoolean "urgent_on_bell" cfg.urgentOnBell}
+
+        ${cfg.optionsExtra}
+
+        [colors]
+        ${optionalString "background" cfg.backgroundColor}
+        ${optionalString "cursor" cfg.cursorColor}
+        ${optionalString "cursor_foreground" cfg.cursorForegroundColor}
+        ${optionalString "foreground" cfg.foregroundColor}
+        ${optionalString "foregroundBold" cfg.foregroundBoldColor}
+        ${optionalString "highlight" cfg.highlightColor}
+
+        ${cfg.colorsExtra}
+
+        [hints]
+        ${optionalString "active_background " cfg.hintsActiveBackgroundColor}
```

Comment:
All of these option strings have an extra space at the end. I guess that is unintentional?